bazel/external/tai.BUILD

```diff
@@ -0,0 +1,46 @@
+package(default_visibility = ["//visibility:public"])
+
+genrule(
+    name = "stub_build",
+    srcs = [":meta_build"],
+    outs = [
+        "libtai.so",
+        ],
+    cmd = "make -C $(location :stub) && cp $(location :stub)/libtai.so \
+    $(@D)",
+    visibility = [ '//visibility:public' ],
+    # TODO: need to find solution how to use tai directories in the 'cmd' script
+    tools = ["stub"]
+)
+
+genrule(
+    name = "meta_build",
```

Comment:
I'd suggest creating a `py_binary` rule for `main.py` so we don't need to install additional python libraries to the base image.

Reply:
Correct me if I'm wrong. By creating a new `py_binary` rule, you mean to add dependencies using `pip_import` (https://github.com/bazelbuild/rules_python#importing-pip-dependencies)?
If yes then we should use a new feature for import pip3 libraries that were recently merged (https://github.com/bazelbuild/rules_python/pull/256). 
And it looks like that our current bazel version does not support their changes. I tried to update our 'io_bazel_rules_python' to the latest version and got this error "io_bazel_rules_python/python/pip.bzl:44:14: unexpected keyword 'doc', for call to function repository_rule(implementation, attrs = None, local = False, environ = [])"

Reply:
Actually Python2 works. You can try using this rules.
See this commit: https://github.com/Yi-Tseng/oopt-tai/commit/8697145db0a6ce47e2e6483680699c8cc2399ce5

Reply:
I tried your example, and faced the following error:

`clang.cindex.TranslationUnitLoadError: Error parsing translation unit.
Target //stratum/hal/bin/dummy:stratum_dummy failed to build`.

Looks like main.py inc/tai.h arg can't be found during 'dummy_switch' compilation.
That is caused by hardcoded paths to the file arguments for 'main.py'.

Your example is great, so I made some [fixes and improvements](https://github.com/Yi-Tseng/oopt-tai/pull/1) for your commit to avoid compilation errors.
And here in Stratum, I have updated bazel make the system to be synced with your repo temporarily.
Is my understanding proper that you might push these changes to the official TAI repo?
If yes, then I will update bazel in Stratum to use TAI official repo once these changes are in there.

Reply:
Please let me know if you are ok with the changes.
If you don’t have any concerns I would like to push these changes to the official TAI repo and update Stratum with the reference to official repo instead of mine.

Reply:
Hi @bohdan-oheruk-plvision, sorry for the delay. I was traveling last week.
The patch from you looks good to me. I think we can use your changes in Stratum before they add Bazel changes to upstream. Also, feel free to push those changes to the official repo.